cloud-azure/src/main/resources/templates/arm-dbstack.ftl

```diff
@@ -155,20 +155,54 @@
             "defaultValue": "${privateEndpointName}",
             "type": "String"
         }
+        <#if dataEncryption == true>
+        , "keyVaultName": {
+              "type": "string",
+              "defaultValue" : "${keyVaultName}",
+              "metadata": {
+                "description": "Key vault name where the key to use is stored"
+              }
+        },
+        "keyVaultResourceGroupName": {
+              "type": "string",
+              "defaultValue" : "${keyVaultResourceGroupName}",
+              "metadata": {
+                "description": "Key vault resource group name where it is stored"
+              }
+        },
+        "keyName": {
+              "type": "string",
+              "defaultValue" : "${keyName}",
+              "metadata": {
+                "description": "Key name in the key vault to use as encryption protector"
+              }
+        },
+        "keyVersion": {
+              "type": "string",
+              "defaultValue" : "${keyVersion}",
+              "metadata": {
+                "description": "Version of the key in the key vault to use as encryption protector"
+              }
+        }
+        </#if>
     },
     "variables": {
         "apiVersion":"2017-12-01",
         "sslEnforcementString":"[if(parameters('useSslEnforcement'), 'Enabled', 'Disabled')]",
         "minimalTlsVersionString":"[if(parameters('useSslEnforcement'), 'TLS1_2', 'TLSEnforcementDisabled')]",
         "geoRedundantBackupString":"[if(parameters('geoRedundantBackup'), 'Enabled', 'Disabled')]",
         "storageAutoGrowString":"[if(parameters('storageAutoGrow'), 'Enabled', 'Disabled')]",
-        "subnetList": "[split(parameters('subnets'),',')]"
+        "subnetList": "[split(parameters('subnets'),',')]",
+        "serverKeyName": "[concat(parameters('keyVaultName'), '_', parameters('keyName'), '_', parameters('keyVersion'))]"
```

Comment:
Not an issue, but a curious question, should we guard this line with either `<#if dataEncryption == true> ,` or `if(parameters('`>